src/redux/productsRedux.js

```diff
@@ -5,9 +5,27 @@ export const getCount = ({ products }) => products.length;
 export const getNew = ({ products }) =>
   products.filter(item => item.newFurniture === true);
 
+export const getAllFavourite = ({ favourite }) => favourite;
+
+export const getCountFavourite = ({ favourite }) => favourite.products.length;
+
+/* action name creator */
+const createActionName = name => `products/favourite/${name}`;
+
+/* action types */
+export const SET_FAVOURITE = createActionName('SET_FAVOURITE');
+
+/* action creator */
+export const changeWishlist = payload => ({ payload, type: SET_FAVOURITE });
+
 /* reducer */
 export default function reducer(statePart = [], action = {}) {
   switch (action.type) {
+    // case SET_FAVOURITE:
+    //   return {
+    //     ...statePart,
+    //     products: [...statePart.products, action.payload],
+    //   };
```

Comment:
Uświadomo sobie czym ten `statePart` jest w Twoim przypadku, jak spojrzymy na inicjalizację magazynu https://github.com/lukaszewczak/WDP-2010-01/blob/master/src/redux/store.js#L14 to widać, żę w tym przypadku `statePart` to tablica wszystkich produktów.

Czyli w ramach ustawinia flagi `"ulubiony"` na produkcie musisz zwrócić nową tablicę w której dla znalezionego po `id` produkcie zmieniasz flagę `"ulubiony"` w taki sposób, że jeżeli miała wartość `false` to będzie miała `true` i jak ma `true` to będzie miała `false`.

Aby stworzyć taką nową tablicę ze zmodyfikowanym produktem możesz użyć np funkcji `map`.

Podsyłam dodatkowe materiały o reducerze, https://redux.js.org/basics/reducers#handling-more-actions